### PR TITLE
Remove `layout_flexbox` function from prelude

### PR DESCRIPTION
**Pull request by Weibye**
# Objective

The prelude contained a single method that has been kept around for backwards compatibility. It is simply a wrapper around `FlexboxAlgorithm::perform_layout()`. The idiomatic way to use taffy now and going forward is to use the `perform_layout()` on the trait so I don't think we need to keep this around for much longer. 

The migration path is also very simple, but it is a breaking change.

## Feedback wanted

Are there any reasons why we would not want this? Or want to delay this? I know #326 might change how end users interact with taffy, and so we might want to bundle these changes together?

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
I like these changes. @nicoburns, what are your plans for the next breaking release? I think we should probably hold off on this PR until we have a better reason for a breaking release.

**Comment by nicoburns** [comment]
> I like these changes. @nicoburns, what are your plans for the next breaking release? I think we should probably hold off on this PR until we have a better reason for a breaking release.

My plan is roughly to include the following:

- https://github.com/DioxusLabs/taffy/pull/413 (non-breaking)
- https://github.com/DioxusLabs/taffy/pull/412 (probably `overflow: hidden` and `overflow: scroll` but not `overflow: auto`)
- Key parts of https://github.com/DioxusLabs/taffy/pull/326 (split into separate PRs)

I'm pretty happy to accept breaking changes into `main` at this point (but agree that we should hold off on actually releasing them). We have the 0.3.x branch that we can use to backport non-breaking bugfixes if need be (@alice-i-cecile - did you ever make those config changes to disable "merge committing" PRs but enable pushing directly to `0.x.y` branches for maintainers?)

---

Finally, I would note that I'm not sure that the current `FlexboxAlgorithm`-like interfaces make sense either, as I've found it awkward to make all of the layout algorithms conform to a single `LayoutAlgorithm` trait. So I would expect there to be further changes to the "low-level" public API prior to a 0.4 release. I think that being the case actually makes now a good time to remove this method though, as we'll be making breaking changes in this area anyway.

File: RELEASES.md
Comment (nicoburns) [inline RELEASES.md:7]:
```suggestion
- `layout_flexbox()` has been removed from the prelude. Use `FlexboxAlgorithm::perform_layout()` instead.
```

**Review by nicoburns — CHANGES_REQUESTED** [review]
Rustfmt and markdown lints need to be fixed.

**Review by nicoburns — APPROVED** [review]

**Pull request merged**
